cloud: Add doc about learning vector index build progress (#18426)
pingcap/docs: tidb-cloud/vector-search-improve-performance.md

The [vector search index](/tidb-cloud/vector-search-index.md) dramatically improves the performance of vector search queries, usually by 10x or more, with a trade-off of only a small decrease of recall rate.

> ## Ensure vector search index is finished

Reviewer: ``` suggestion
## Ensure vector indexes are fully built
```

> The vector indexer runs asynchronously. Before all vector data is indexed, the vector search performance is suboptimal. See [View index build progress](/tidb-cloud/vector-search-index.md#view-index-build-progress) to learn how to check the index build progress.

Reviewer: ```suggestion
Vector indexes are built asynchronously. Until all vector data is indexed, vector search performance is suboptimal. To check the index build progress, see [View index build progress](/tidb-cloud/vector-search-index.md#view-index-build-progress).
```
